// This file is part of KWIVER, and is distributed under the
// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.

syntax = "proto2";

import "detected_object_type.proto";
import "bounding_box.proto";
import "geo_point.proto";
import "point.proto";

package kwiver.protobuf;

message detected_object {

  required double confidence = 1;
  required bounding_box bbox = 2;
  optional detected_object_type classifications = 3;
  optional int64 index = 4;
  optional string detector_name = 5;
  repeated string notes = 6;
  map< string, point_d> keypoints  = 7;
  optional geo_point geopoint = 8;

  // Currently skipping image chip and descriptor
  //+TBD
}
